### ISSUE TITLE
Cannot use --init_model_path

### ISSUE THREAD

**Issue by ghost** [PaddlePaddle/Paddle#1738]
I try to use the NMT model you trained (fr->en). 
When i use --init_model_path=<your_pretrained_model>, I get the following error:

I0401 16:41:14.938364  3162 TrainerInternal.cpp:165]  Batch=2 samples=100 AvgCost=nan CurrentCost=nan Eval: classification_error_evaluator=0.696074  CurrentEval: classification_error_evaluator=1 

The command line is 
paddle train \
--config='translation/train.conf' \
--save_dir='translation/model/wmt14_model' \
--use_gpu=1 \
--num_passes=16 \
--show_parameter_stats_period=1 \
--trainer_count=1 \
--log_period=1 \
--dot_period=5 \
--init_model_path=model/wmt14_model/pass-00012 \
2>&1 | tee 'translation/train.log'

Can any one help me?

Thanks

**Comment by livc** [comment]
Recently we have updated the Python API(v2). So could you please pull the latest repo and try it again ? 
https://github.com/PaddlePaddle/book/blob/develop/07.machine_translation/README.en.md

Thanks.

**Comment by helinwang** [comment]
@KeepLearning12138 Please refer to here as how to run PaddlePaddle: https://github.com/PaddlePaddle/Paddle/blob/develop/doc/getstarted/build_and_install/docker_install_en.rst

**Comment by KeepLearning12138** [comment]
@livc @helinwang Thanks a lot for your reply. According to your comments, I installed docker-image on a isolate server and run the demo code shown in ch.7 of the paddle-book. Then, I run the train.py code and get the following results: 

Pass 0, Batch 0, Cost 164.959180, {'classification_error_evaluator': 1.0}
.........
Pass 0, Batch 10, Cost 241.103516, {'classification_error_evaluator': 0.9316239356994629}
.........
Pass 0, Batch 20, Cost 341.873877, {'classification_error_evaluator': 0.9216867685317993}
.........
Pass 0, Batch 30, Cost 189.436011, {'classification_error_evaluator': 0.9130434989929199}
.........
Pass 0, Batch 40, Cost 209.991748, {'classification_error_evaluator': 0.9509803652763367}
.........
Pass 0, Batch 50, Cost 267.384741, {'classification_error_evaluator': 0.8999999761581421}
.........
Pass 0, Batch 60, Cost 213.720215, {'classification_error_evaluator': 0.942307710647583}
.........
Pass 0, Batch 70, Cost 195.130457, {'classification_error_evaluator': 0.9473684430122375}
.........
Pass 0, Batch 80, Cost 158.140320, {'classification_error_evaluator': 0.9610389471054077}
.........
Pass 0, Batch 90, Cost nan, {'classification_error_evaluator': 1.0}
.........
Pass 0, Batch 100, Cost nan, {'classification_error_evaluator': 1.0}
.........
Pass 0, Batch 110, Cost nan, {'classification_error_evaluator': 1.0}

Can we directly use "train.py"? 

Best Wishes

**Comment by helinwang** [comment]
@KeepLearning12138 Sure, please see the link that I posted in the previous comment: https://github.com/PaddlePaddle/Paddle/blob/develop/doc/getstarted/build_and_install/docker_install_en.rst. The specific line is at:

```bash
docker run --rm -v ~/workspace:/workspace paddlepaddle/paddle:0.10.0rc2 python /workspace/train.py
```
The above comment mounts `~/workspace` into `/workspace` inside the docker image, and run `python /workspace/train.py` when starting the image.
You can copy the `train.py` from https://github.com/PaddlePaddle/book into `~/workspace`.

**Comment by KeepLearning12138** [comment]
@helinwang Thanks for the quick reply. Actually, I follow your suggestion. What i mean is that: although I can run the "train.py", my code is easily trapped in "NaN". When i set the learning rate as "0.0", the NaN disappears. I just want to confirm where the error comes from.

**Comment by helinwang** [comment]
Good to know that you are experimenting around! However setting learning rate to 0.0 the network will stop to learn anything.
I think this PR is trying to fix the problem you are encountering: https://github.com/PaddlePaddle/book/pull/234. What it does is trying to add regularization to the network so that weights will not go wild.
Could you check if your `train.py` contains the changes in the PR, if so, you can even make the regularization rate higher.

**Comment by KeepLearning12138** [comment]
Thanks for the reply. 
I do not think the L2Reg is a key problem to neural machine translation. Because according to my previous experience, for a 1-layer GRU, it is OK not using weight decay. What I need to solve is: since you have already release a pretrained model (fr->en), I want to use your model as a warm start. After passing 100 samples, AvgCost=nan. Even if setting lr=0, the same bug appears. I am really curious about this problem. Can you solve it?

**Comment by luotao1** [comment]
@KeepLearning12138 after #1750, you can use 
```
parameters = paddle.dataset.wmt14.model()
```
instead of 
```
parameters = paddle.parameters.create(cost)
```
to do a warm start.

**Comment by KeepLearning12138** [comment]
Still not worked yet. Many NaN are detected. Only if i set batch_size as 1, the NaN seems disappearing.

**Comment by alvations** [comment]
I'm having similar issue with cost going to `nan` when training using the the new https://github.com/PaddlePaddle/book/blob/develop/07.machine_translation/README.en.md

**Comment by livc** [comment]
@alvations How about trying to reduce the learning rate?

**Comment by alvations** [comment]
I've tried the older setup which used to work in v0.8.0 and v0.9.0:

```
    batch_size = 50,
    learning_rate = 5e-4
    L2Regularization(rate=8e-4)
```

The current default from https://github.com/PaddlePaddle/book/blob/develop/08.machine_translation/train.py#L143 (as below) is frustratingly slow. 

```
    batch_size = 5,
    learning_rate = 5e-5
    L2Regularization(rate=8e-4)
```


What are the changes to the optimizer such that the old settings doesn't work any more?

**Comment by lcy-seso** [comment]
gradient clipping has been fixed in the current develop branch. And we also fix a terrible bug of sequnece_softmax. The NMT training in 0.10.0 has been fixed now.

I close this issue due to inactivity, please feel free to reopen it.
